checkdeps.py

```diff
@@ -199,13 +231,36 @@ def testCompiler():
 if prereqs:
     mandatory = list(x for x in prereqs if "optional" not in PACKAGES[x] or not PACKAGES[x]["optional"])
     optional = list(x for x in prereqs if "optional" in PACKAGES[x] and PACKAGES[x]["optional"])
+
     if mandatory:
         print "Missing mandatory dependencies: %s" % (" ".join(mandatory))
     if optional:
         print "Missing optional dependencies: %s" % (" ".join(optional))
         for package in optional:
             print PACKAGES[package].get('description')
 
+# Install the system dependencies of optional extras_require components
+OPSYS = detectOS()
+CMD = PACKAGE_MANAGER[OPSYS] if OPSYS in PACKAGE_MANAGER else 'UNKNOWN_INSTALLER'
+for lhs, rhs in EXTRAS_REQUIRE.items():
+    if rhs and any([
+        EXTRAS_REQUIRE_DEPS[x][OPSYS]
+        for x in rhs
+        if x in EXTRAS_REQUIRE_DEPS
+    ]):
+        rhs_cmd = ''.join([
+            CMD,
+            ' ',
+            ' '.join([
+                ''. join([
+                    xx for xx in EXTRAS_REQUIRE_DEPS[x][OPSYS]
+                ])
+                for x in rhs
+                if x in EXTRAS_REQUIRE_DEPS
+            ]),
+        ])
+        print "Optional dependency `pip install .[{}]` would require `{}` to be run as root".format(lhs, rhs_cmd)
```

Comment:
I got lost in this expression ):

Reply:
Also I've used [print](https://docs.python.org/3/library/functions.html#print) which is available in python 2.7 because it's a slow preparation to python3.

Reply:
OK, I will unroll it in the next commit.

Reply:
Ah, already merged. Can you open an issue please and I'll get round to it.

Reply:
That's a matter of taste. I won't mind if @PeterSurda accepts.